spec/services/media_storage/azure_adapter_spec.rb

```diff
@@ -0,0 +1,211 @@
+require 'spec_helper'
+
+RSpec.describe MediaStorage::AzureAdapter do
+  let(:storage_account_name) { 'tiny-watermelons' }
+  let(:container) { 'magic-container' }
+  let(:opts) do
+    {
+      azure_prefix: 'test-uploads.zooniverse.org/',
+      azure_storage_account: storage_account_name,
+      azure_storage_access_key: 'fake',
+      azure_storage_container: container
+    }
+  end
+  let(:adapter) { described_class.new(opts) }
+  let(:uri_regex) { /\A#{URI::DEFAULT_PARSER.make_regexp}\z/ }
+  let(:uuid_v4_regex) { /[a-f0-9]{8}\-[a-f0-9]{4}\-4[a-f0-9]{3}\-(8|9|a|b)[a-f0-9]{3}\-[a-f0-9]{12}/ }
+
+  let(:signer) { instance_double('Azure::Storage::Common::Core::Auth::SharedAccessSignature') }
+  let(:blob_client) { instance_double('Azure::Storage::Blob::BlobService') }
+
+  before do
+    allow(Azure::Storage::Common::Core::Auth::SharedAccessSignature).to receive(:new) { signer }
+    allow(Azure::Storage::Blob::BlobService).to receive(:create) { blob_client }
+  end
+
+  context 'when opts are passed to the initializer' do
+    it 'uses default expiration values when no expiration values are passed' do
+      default = MediaStorage::AzureAdapter::DEFAULT_EXPIRES_IN
+      expect(adapter.get_expiration).to eq(default)
+      expect(adapter.put_expiration).to eq(default)
+    end
+
+    it 'defaults to current rails environment for the container name when no container is given' do
+      adapter = described_class.new(opts.except(:azure_storage_container))
```

Comment:
I don't think we want this as we want explicitly specify the desired container and have these pre-setup with the relevant permissions. 
rather we should raise or error if the container isn't setup correctly

Reply:
@camallen what about the prefix? I know we have the prefix set up in AWS to default to the current env when nothing is passed in; I'm not sure why we do this in AWS, want to make sure I understand before rolling out the azure adapter with the same setup.